app/helpers/application_helper.rb

```diff
@@ -276,4 +276,148 @@ def sign_in_path
   def unread_notifications(user)
     Comment.accessible_by(current_ability).find_since_last_login(user)
   end
+
+  # Recieves a PaperTrail::Version object
+  # Outputs the list of attributes that were changed in the version (ignoring changes from one blank value to another)
+  # Eg: If version.changeset = '{"title"=>[nil, "Premium"], "description"=>[nil, "Premium = Super cool"], "conference_id"=>[nil, 3]}'
+  # Output will be 'title, description and conference'
+  def updated_attributes(version)
+    version.changeset.
+      reject{ |_, values| values[0].blank? && values[1].blank? }.
+      keys.map{ |key| key.gsub('_id', '').tr('_', ' ')}.join(', ').
+      reverse.sub(',', ' dna ').reverse
+  end
+
+  def name_or_username(user_id)
+    user = User.find_by(id: user_id)
+    return nil unless user
+    if user.name.blank?
+      user.username
+    else
+      user.name
+    end
+  end
+
+  def change_creator_link(user_id)
+    if user_id
+      link_to name_or_username(user_id), admin_user_path(id: user_id)
+    else
+      'Someone (probably via the console)'
+    end
+  end
+
+  def get_version_object(version)
+    version.item || version.reify || version.next.reify
+  end
+
+  def event_change_description(version)
+    case
+    when version.event == 'create' then 'submitted new'
+
+    when version.event == 'destroy' then 'deleted'
+
+    when version.changeset['state']
+      case version.changeset['state'][1]
+      when 'unconfirmed' then 'accepted'
+      when 'withdrawn' then 'withdrew'
+      when 'canceled', 'rejected', 'confirmed' then version.changeset['state'][1]
+      when 'new' then 'resubmitted'
+      end
+
+    when version.changeset['start_time'] && version.changeset['start_time'][0].nil?
+      'scheduled'
+
+    when version.changeset['start_time'] && version.changeset['start_time'][1].nil?
+      'unscheduled'
+
+    else
+      "updated #{updated_attributes(version)} of"
+    end
+  end
+
+  def users_role_change_description(version)
+    user_id = get_version_object(version).user_id
+    if version.event == 'create'
+      "added #{name_or_username(user_id)} to"
+    else
+      "removed #{name_or_username(user_id)} from"
+    end
+  end
+
+  def subscription_change_description(version)
+    user_id = get_version_object(version).user_id
+    user_id = nil if user_id.to_s == version.whodunnit
+    # FIXME
+    version.event == 'create' ? "subscribed #{name_or_username(user_id)} to" : "unsubscribed #{name_or_username(user_id)} from"
+  end
+
+  def registration_change_description(version)
+    user_id = get_version_object(version).user_id
+    if user_id.to_s == version.whodunnit
+      case version.event
+      when 'create' then 'registered to'
+      when 'update' then "updated #{updated_attributes(version)} of  #{name_or_username(user_id)}'s registration for"
```

Comment:
In this case that the registration.user is making the change we can rephrase this, as in create and destroy actions.